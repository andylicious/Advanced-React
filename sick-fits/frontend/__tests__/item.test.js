import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import ItemComponent from '../components/Item'

const mockItem = {
  id: '12345abcdef',
  title: 'A pair of shoes',
  price: 50000,
  description: 'wear them on feet',
  image: 'dog.jpg',
  largeImage: 'largeDog.jpg',
}

describe('<Item />', () => {
  it('renders and matches snapshot', () => {
    const wrapper = shallow(<ItemComponent item={mockItem} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  // let wrapper
  // beforeEach(() => {
  //   wrapper = shallow(<ItemComponent item={mockItem} />)
  // })

  // it('renders the image properly', () => {
  //   const img = wrapper.find('img')
  //   expect(img.props().src).toBe(mockItem.image)
  //   expect(img.props().alt).toBe(mockItem.title)
  // })

  // it('renders and displays properly', () => {
  //   const pTag = wrapper.find(PriceTag)
  //   expect(pTag.children().text()).toBe('$500')
  //   expect(wrapper.find('ForwardRef(Title) Link a').text()).toBe(mockItem.title)
  // })

  // it('renders out the buttons properly', () => {
  //   const buttonList = wrapper.find('.buttonList')
  //   expect(buttonList.children()).toHaveLength(3)
  //   expect(buttonList.find('Link').exists()).toBe(true)
  //   expect(buttonList.find('AddToCart')).toHaveLength(1)
  //   expect(buttonList.find('DeleteItem')).toHaveLength(1)
  // })
})
