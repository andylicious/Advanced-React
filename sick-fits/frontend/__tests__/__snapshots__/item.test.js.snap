// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches snapshot 1`] = `
<ForwardRef(ItemStyles__Item)>
  <img
    alt="A pair of shoes"
    src="dog.jpg"
  />
  <ForwardRef(Title)>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "12345abcdef",
          },
        }
      }
    >
      <a>
        A pair of shoes
      </a>
    </Link>
  </ForwardRef(Title)>
  <ForwardRef(PriceTag)>
    $500
  </ForwardRef(PriceTag)>
  <p>
    wear them on feet
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "12345abcdef",
          },
        }
      }
    >
      <a>
        Edit ðŸ–‹
      </a>
    </Link>
    <AddToCart
      id="12345abcdef"
    />
    <DeleteItem
      id="12345abcdef"
    >
      Delete
    </DeleteItem>
  </div>
</ForwardRef(ItemStyles__Item)>
`;
